api/src/middleware/competenciesRouter.ts

```diff
@@ -48,4 +50,47 @@ competenciesRouter.post('/', async (req, res, next) => {
   res.status(201).json(itemEnvelope(competency));
 });
 
+competenciesRouter.put('/:id', async (req, res, next) => {
+  const { principalId } = req.session;
+  const { id } = req.params;
+
+  const competencyId = Number(id);
+  if (!Number.isInteger(competencyId) || competencyId < 1) {
+    next(new BadRequestError(`"${id}" is not a valid competency id.`));
+    return;
+  }
+
+  let isAuthorized;
+  try {
+    isAuthorized = await authorizeCompetencyUpdate(principalId, competencyId);
+  } catch (error) {
+    next(error);
+    return;
+  }
+  if (!isAuthorized) {
+    next(
+      new NotFoundError(
+        `A competency with the id "${competencyId}" could not be found.`
+      )
+    );
+    return;
+  }
+  const { label, description } = req.body;
+  if (typeof label !== 'string') {
+    next(new ValidationError('label must be a string'));
+    return;
+  }
+  if (typeof description !== 'string') {
+    next(new ValidationError('description must be a string'));
+    return;
+  }
+  try {
+    await updateCompetency(competencyId, label, description);
+  } catch (error) {
+    next(error);
+    return;
+  }
+  res.status(200).json(itemEnvelope({ id: competencyId }));
```

Comment:
The default status when json is sent is `200`. Is it necessary to call `.status(200)` here?

Reply:
yes indeed. simplified to `res.json(...)`